crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1703,37 +1703,47 @@ where
                         // if we're currently syncing and the inserted block is the targeted
                         // FCU head block, we can try to make it canonical.
                         if block_hash == target.head_block_hash {
-                            if let Err((_hash, error)) =
-                                self.try_make_sync_target_canonical(block_num_hash)
-                            {
-                                if error.is_fatal() {
-                                    let response = Err(BeaconOnNewPayloadError::Internal(
-                                        Box::new(error.clone()),
-                                    ));
-                                    let _ = tx.send(response);
-                                    return Err(RethError::Canonical(error))
-                                }
-
-                                // If we could not make the sync target block canonical,
-                                // we should return the error as an invalid payload status.
-                                let status = Ok(PayloadStatus::new(
-                                    PayloadStatusEnum::Invalid {
-                                        validation_error: error.to_string(),
-                                    },
-                                    // TODO: return a proper latest valid hash
-                                    // See: <https://github.com/paradigmxyz/reth/issues/7146>
-                                    self.forkchoice_state_tracker.last_valid_head(),
-                                ));
-                                let _ = tx.send(status);
-                                return Ok(EngineEventOutcome::Processed)
-                            }
+                            let previous_action = self.blockchain_tree_action.replace(
```

Comment:
I assume eventually this will be converted into a Vec or channel?

Reply:
exactly! but baby steps for now